### ISSUE TITLE
How to change PHP memory limit

### ISSUE THREAD

**Issue by theRAGEhero** [YunoHost-Apps/wordpress_ynh#205]
Hello,
how can I change the PHP memory limit?
If I do it from wp-config.php it breaks the website.
Is there another way? I already set the WordPress setting ynh to High, as [from this thread](https://forum.yunohost.org/t/apps-keep-restarting/21712).
Thank you

**Comment by theRAGEhero** [comment]
Ok, I've found a solution.

https://forum.yunohost.org/t/php-memory-limit-en-fr/3836/14

Don't forget to use the right PHP version (8.0 at the moment), and to specify your __n, like wordpress__3

**Comment by dankerthrone** [comment]
for any idiots like me finding this in the future: that means you need to edit /etc/php/xxx/fpm/pool.d/wordpress__X.conf if you have multiple instances running 😆

**Comment by alexAubin** [comment]
Similarly to https://github.com/YunoHost-Apps/wordpress_ynh/issues/189#issuecomment-3671732896 , 

you should be able to tweak the value using

```
sudo yunohost app setting wordpress php_memory_limit -v 512M
sudo yunohost app upgrade wordpress --force
```

(and contrarily to the message about upload max filesize, there's no relation to the nginx part this time)

**Comment by dankerthrone** [comment]
yes, but keep in mind that similarly to #189 this manual overwrite will probably be deleted by the next wordpress_ynh update for some reason

**Comment by alexAubin** [comment]
It's not a "manual overwrite" ... Defining the setting will [precisely affect *the autogenerated conf from YunoHost for this app*](https://github.com/YunoHost/yunohost/blob/704cf0872c79e2f018e1b52d4ab5c7975c439813/helpers/helpers.v2.1.d/php#L118) (in this code/template snippet, the value used for `__PHP_MEMORY_LIMIT__` is **coming from the `php_memory_limit` setting**, or 128M by default if the setting doesn't exists).

The very existence of this setting is to allow reliable, ad-hoc customization by admins...

**Comment by bernhardkaindl** [comment]
JFYI, the `testing` branch https://github.com/YunoHost-Apps/wordpress_ynh/compare/testing (scroll to the bottom to find the current changes to `scripts/install`=`) has increased limits for new installations:

```ml
ynh_app_setting_set_default --key=php_memory_limit --value=768M
ynh_app_setting_set_default --key=php_upload_max_filesize --value=512M
ynh_app_setting_set_default --key=php_post_max_size --value=512M
ynh_app_setting_set_default --key=php_max_file_uploads --value=512M
```

When testing is successful, it would become the new default WordPress application.

Note: If you'd test to change WordPress URLs, I've an incremental fix on top of the official testing branch here:

```ml
sudo yunohost app install -l New-Wordpress -a "domain=domain.tld&path=/&init_main_permission=visitors" https://github.com/Akzente-Training/wordpress_ynh/tree/testing
```
